quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresEncoders.scala

```diff
@@ -0,0 +1,73 @@
+package io.getquill.context.ndbc
+
+import java.time.{ LocalDate, LocalDateTime, ZoneOffset }
+import java.util.{ Date, UUID }
+
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+import io.getquill.dsl.CoreDsl
+import io.trane.ndbc.PostgresPreparedStatement
+
+trait LowPriorityPostgresImplicits {
+  this: CoreDsl =>
+
+  implicit def mappedEncoder[I, O](implicit mapped: MappedEncoding[I, O], e: BaseEncoder[O]): BaseEncoder[I] =
+    mappedBaseEncoder(mapped, e)
+}
+
+trait PostgresEncoders extends LowPriorityPostgresImplicits with io.getquill.dsl.LowPriorityImplicits {
+  this: NdbcContext[_, _, PostgresPreparedStatement, _] =>
+
+  type Encoder[T] = BaseEncoder[T]
+
+  protected val zoneOffset: ZoneOffset
+
+  def encoder[T, U](f: PostgresPreparedStatement => (Int, U) => PostgresPreparedStatement)(implicit ev: T => U): Encoder[T] =
+    (idx, v, ps) =>
+      if (v == null) ps.setNull(idx)
+      else f(ps)(idx, v)
+
+  def arrayEncoder[T, U: ClassTag, Col <: Seq[T]](f: PostgresPreparedStatement => (Int, Array[U]) => PostgresPreparedStatement)(ev: T => U): Encoder[Col] =
+    (idx, v, ps) =>
+      if (v == null) ps.setNull(idx)
+      else f(ps)(idx, v.map(ev).toArray[U])
+
+  implicit override def anyValMappedEncoder[I <: AnyVal, O](implicit mapped: MappedEncoding[I, O], encoder: Encoder[O]): Encoder[I] = mappedEncoder
+
+  implicit def optionEncoder[T](implicit e: Encoder[T]): Encoder[Option[T]] =
+    (idx, v, ps) =>
+      v match {
+        case None    => ps.setNull(idx)
```

Comment:
V will not be null here when the cell is null?

Reply:
Well spotted :)